library/Icingadb/Widget/Detail/PerfDataTable.php

```diff
@@ -114,13 +119,30 @@ public function assemble()
                 }
 
                 foreach ($perfdata->toArray() as $column => $value) {
+                    $content = $value ? Text::create($value) : new EmptyState(t('None', 'value'));
+                    if ($column === 'label' && ! $perfdata->isValid()) {
```

Comment:
Handle this like if the perfdata is visualizable. (Outside this loop) The icon should be in the first column, not where the label is.

Reply:
Note that `containsSparkline` is effectively a *"Is there a perfdata with an icon?"*. So set it also to true if some perfdata is invalid.